### ISSUE TITLE
Feature Request: SPICE protocol

### ISSUE THREAD

**Issue by scardella** [mRemoteNG/mRemoteNG#1739]
I'd love to be able to use mRemoteNG to handle libvirt/QEMU SPICE protocol.  That would allow connections to libvirt/QEMU VMs even if there was some reason that networking was not working within the VM.  OS installs, upgrades, etc.  Oddball OSes that you don't know whether there's remote access.  Or, diagnosing networking issues in the VM.

It seems like it would slot in well with the mRemoteNG's ethos.

**Comment by Benchwork** [comment]
I agree that it would be nice, however at least in windows the standalone SPICE installer is 80Mb thats a lot to add to the total size of mRemoteNG for one protocol.   have your tried to install it and get it to work with external tools?  if so can you share how you have it setup?

**Comment by ndroftheline** [comment]
this wrapper appears to work brilliantly for me, pointing mremoteNG at this wrapper to connect to a VM on my proxmox host using SPICE protocol from a Windows machine:

https://forum.proxmox.com/threads/remote-spice-access-without-using-web-manager.16561/post-262411

that is, download executable from https://moxhamconsultants.com/proxmox/ProxmoxSpiceLauncher.zip

command syntax is 

ProxmoxSpiceLauncher.exe host=x.x.x.x port=8006 username=root@pam password=xxx node=pve vm=121 viewer="C:\Program Files\VirtViewer v8.0-256\bin\remote-viewer.exe"

code dumped here: https://github.com/toby11/windows-promox-spice-viewer 

there is also another project that appears to have it a little more together here that might act as a guide: https://github.com/Corsinvest/cv4pve-pepper

good luck, thanks

**Comment by scardella** [comment]
Micah,

Did it open up in a mRemote tab, or did it pop up in an external window?

Steve Cardella
Solution Consultant | BlueGranite, Inc.<https://www.blue-granite.com/>
m: (504) 669-7822 | p: (877) 817-0736 ext. 735
scardella@blue-granite.com<mailto:scardella@blue-granite.com>

**Comment by ndroftheline** [comment]
Opens in an external window for me; that's how I prefer it anyway. I'm not
sure that's configurable or not.

**Comment by pricerc** [comment]
It might be worth looking at the home of SPICE:
https://gitlab.freedesktop.org/spice/spice-gtk

and at the viewer that they recommend:
https://gitlab.com/virt-viewer/virt-viewer

**Comment by NoxiousPluK** [comment]
+1 for SPICE support since TrueNAS Scale now also uses this

**Comment by c-amie** [comment]
+1 also

**Comment by jonnytest1** [comment]
+1 (if download size is relevant maybe and opt in plugin)

**Comment by lpet** [comment]
+1 as Proxmox is using Spice as well

**Comment by jijtech** [comment]
+1 (Proxmox&TrueNas use spice)

**Comment by saniyo** [comment]
+1

**Comment by robertpopa22** [comment]
Triage refresh (P2): this issue has been inactive for a long time. Please retest on the latest build and share reproducible steps, expected vs actual behavior, and logs/screenshots. If there is no confirmation/update, maintainers can close it as stale or not reproducible.

**Comment by jonnytest1** [comment]
wait my mremoteng says the current version is 1.76.20 but the most recent one available is 1.76.11 - how can i be more current than the lastest one 😅 

(also the website over at https://mremoteng.org/ is NOT really usable 
- cant refresh on subpaths
- downloads page is empty
- header is hiding content
- "You can't really contact us, because this is a demo. But if you could, this is where the contact form or details would go!"
 )

**Comment by simonai1254** [comment]
> wait my mremoteng says the current version is 1.76.20 but the most recent one available is 1.76.11 - how can i be more current than the lastest one 😅 
> 
> (also the website over at https://mremoteng.org/ is NOT really usable 

Yeah the website needs some love, and the updater check is broken (known issue, but don't have the reference handy). The latest (preview) releases can be obtained over at https://github.com/mRemoteNG/mRemoteNG/releases

**Comment by Kvarkas** [comment]
let me clarify this a bit, so we have 3 streams: prod ready, preview, and night builds
older code is already legacy from all meanings, technically all features fixes are coming into 1.78.2 version - what will be a release candidate (moved from 3rd stream into 2nd) web site was modified but developer get lost, and currently over busy to be able to fix all issues we have regarding this but don't worry nothing is forgotten, and changes will be applied soon
